src/Core/Query/AbstractResponseParser.php

```diff
@@ -50,30 +50,6 @@ public function convertToKeyValueArray(array $data): array
         return $result;
     }
 
-    /**
-     * Parses header data (if available) and adds it to result data.
-     *
-     * @param array $data
-     * @param array $result
-     *
-     * @return array
-     */
-    public function addHeaderInfo(array $data, array $result): array
```

Comment:
I wonder if that change breaks backward compatibility. If I read the PR correctly, status and queryTime have to be accessed differently now by the client.

Reply:
You can still call `->getStatus()` and `->getQueryTime()` on the results that already had those methods. With my changes, you can now call it for every query type's result, not just the ones it happened to be implemented for.

AFAIK it doesn't break BC. The changes I had to make to the unit tests were only for how a result stores this info now, and those are `protected` members so that doesn't matter to the client. If you do find a way to break BC, let me know of course!

Reply:
One way to check this is `examples/1.3-basic-update.php`. It still shows the status and query time without modifications.

Reply:
@mkalkbrenner I went over it again and I get your concern now. I was solely focused on status and queryTime so I'm confident I didn't break those.

Now I'm wondering if we use other parameters from the responseHeader. If I broke something there, it wouldn't have had test coverage though or I would have been aware. I'll convert this PR to a draft until I've had the time to doublecheck this to make sure.

Reply:
In Drupal we have a debug mode for solarium queries. I should check if it still works.

Reply:
Turns out I did break something: if a reponse parser "cherry picked" from the response data, it ignored the response header. This wasn't caught in any of the unit/integration tests and of course I happened to only use query types where this wasn't an issue for my manual verification 🙄. Fixed that and added a unit test.

I also sifted through the Solr source code for every parameter that can be added to `responseHeader`. This is the complete list:
- status
- QTime
- handler
- params
- partialResults
- segmentTerminatedEarly
- zkConnected
- warnings
- rf
- msg
- errors
- maxErrors

Of these, only `status` and `QTime` are always set, and those are also the only ones that were captured by `addHeaderInfo()`. Others depend on handler and some on cloud mode too. `Ping\Result::getZkConnected()` demonstrates how it's trivial to add support for those.